Bottom: 3797e5176c4f13f270831a7e6a9682b1fbd4adc3
Top:    f3b98ea862ba3ec6d38469f3803a2b0f54c6591b
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-01-09 18:01:19 +0800

tun: prefetch skb->data before transport header probing

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 0986413..75f4116 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -72,6 +72,7 @@
 #include <linux/seq_file.h>
 #include <linux/uio.h>
 #include <linux/skb_array.h>
+#include <linux/prefetch.h>
 
 #include <linux/uaccess.h>
 
@@ -1279,6 +1280,7 @@ static ssize_t tun_get_user(struct tun_struct *tun, struct tun_file *tfile,
 		uarg->callback(uarg, false);
 	}
 
+	prefetchw(skb->data);
 	skb_reset_network_header(skb);
 	skb_probe_transport_header(skb, 0);
